lib/matplotlib/tests/test_path.py

```diff
@@ -49,6 +64,19 @@ def test_contains_points_negative_radius():
     np.testing.assert_equal(result, [True, False, False])
 
 
+def test_exact_extents_cubic():
+    # notice that if we just looked at the control points to get the bounding
+    # box of each curve, we would get the wrong answers. For example, for
+    # hard_curve = Path([[0, 0], [1, 0], [1, 1], [0, 1]],
+    #                   [Path.MOVETO, Path.CURVE4, Path.CURVE4, Path.CURVE4])
+    # we would get that the extents area (0, 0, 1, 1). This code takes into
+    # account the curved part of the path, which does not typically extend all
+    # the way out to the control points
+    for extents, path in _test_curves:
```

Comment:
This can be done with `pytest.mark.parametrize` on the function, which also means the definitions don't need to be far away at the top of the file.

Reply:
.... or the definitions need not be at the top of the file anyhow ;-)

Reply:
But they will eventually be anyway, since we're also going to use the same curves to test that the area is computed correctly, so it will be natural for at least the "path" definitions to live at the top of the file. and at that point, putting their extents, area (length, center of mass, etc) separately at the point of the function that computes them seems....ugly, since then the precomputed attributes are far from the definition of the object that they are precomputed for

Reply:
But since we don't merge changes based on non-approved future code I guess I'll move the definitions closer, use pytest's parametrize (although, to be honest, I think it's less clear than just having a for loop...) and worry about the area/length/etc tests when those are getting reviewed.